src/feature/flow_chart_panel/components/NodeWrapper.tsx

```diff
@@ -1,51 +1,43 @@
-import React, { useEffect, useState } from "react";
-import { CustomNodeProps } from "../types/CustomNodeProps";
+import { Box, Text, createStyles } from "@mantine/core";
 import { useFlowChartState } from "@src/hooks/useFlowChartState";
 import { useSocket } from "@src/hooks/useSocket";
-import { Box, createStyles, Text } from "@mantine/core";
-import NodeEditButtons from "./node-edit-menu/NodeEditButtons";
+import React, { useEffect, useState } from "react";
+import { CustomNodeProps } from "../types/CustomNodeProps";
 
 const NodeWrapper = ({
   data,
   children,
 }: CustomNodeProps & {
   children: React.ReactNode;
 }) => {
-  const { failedNode, setNodes, setEdges } = useFlowChartState();
+  const { setIsEditMode } = useFlowChartState();
+  const { failedNode } = useFlowChartState();
   const { states } = useSocket();
   const [runError, setRunError] = useState<{
     message: string;
     show: boolean;
   } | null>(null);
 
-  const handleNodeRemove = (nodeId: string) => {
-    setNodes((prev) => prev.filter((node) => node.id !== nodeId));
-    setEdges((prev) =>
-      prev.filter((edge) => edge.source !== nodeId && edge.target !== nodeId)
-    );
-  };
-
   useEffect(() => {
+    console.log("15");
     if (failedNode === data.id) {
       setRunError({
-        message: states?.failureReason!,
+        message: states?.failureReason ?? "Unknown failure reason",
         show: false,
       });
     }
     return () => {
       setRunError(null);
     };
-  }, [failedNode, data, states?.failureReason]);
+  }, [failedNode, states?.failureReason]);
+
   return (
-    <Box data-testid="node-wrapper" pos="relative">
+    <Box
+      data-testid="node-wrapper"
+      pos="relative"
+      onClick={() => setIsEditMode(true)}
+    >
       {runError && <ErrorPopup message={runError.message} />}
-      {data.selected && (
```

Comment:
X icon was removed because putting delete logic in nodes caused performance issues (created dependency on nodes state in each node). I removed the icon because the functionality was gone, is that ok? (plus it looks weird on non rectangular nodes such as arithmetic)

Reply:
It's not a good experience from a user's perspective I beleive, there should be a clickable button/icon for deleting a node instead of only with the `Backspace` key. What do you think, @jackparmer @itsjoeoui ?

Reply:
Though I understand the performance POV, but I think the X icon is still needed for a better user experience.